src/components/application_manager/rpc_plugins/app_service_rpc_plugin/src/commands/mobile/get_app_service_data_request.cc

```diff
@@ -0,0 +1,107 @@
+ LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "app_service_rpc_plugin/commands/mobile/get_app_service_data_request.h"
+#include "app_service_rpc_plugin/app_service_app_extension.h"
+#include "application_manager/application_impl.h"
+#include "application_manager/message_helper.h"
+#include "application_manager/rpc_service.h"
+#include "interfaces/MOBILE_API.h"
+#include "smart_objects/enum_schema_item.h"
+
+namespace app_service_rpc_plugin {
+using namespace application_manager;
+namespace commands {
+
+GetAppServiceDataRequest::GetAppServiceDataRequest(
+    const application_manager::commands::MessageSharedPtr& message,
+    ApplicationManager& application_manager,
+    app_mngr::rpc_service::RPCService& rpc_service,
+    app_mngr::HMICapabilities& hmi_capabilities,
+    policy::PolicyHandlerInterface& policy_handler)
+    : CommandRequestImpl(message,
+                         application_manager,
+                         rpc_service,
+                         hmi_capabilities,
+                         policy_handler)
+    , plugin_(NULL) {
+  auto plugin = (application_manager.GetPluginManager().FindPluginToProcess(
+      mobile_apis::FunctionID::GetAppServiceDataID,
+      app_mngr::commands::Command::CommandSource::SOURCE_MOBILE));
+  if (plugin) {
+    plugin_ = dynamic_cast<AppServiceRpcPlugin*>(&(*plugin));
+  }
+}
+
+GetAppServiceDataRequest::~GetAppServiceDataRequest() {}
+
+void GetAppServiceDataRequest::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  std::string service_type =
+      (*message_)[strings::msg_params][strings::service_type].asString();
+
+  bool subscribe = false;
+  if ((*message_)[strings::msg_params].keyExists(strings::subscribe)) {
+    subscribe = (*message_)[strings::msg_params][strings::subscribe].asBool();
+  }
+
+  ApplicationSharedPtr app = application_manager_.application(connection_key());
+
+  if (subscribe) {
+    auto& ext = AppServiceAppExtension::ExtractASExtension(*app);
+    ext.SubscribeToAppService(service_type);
+  }
+
+  SendProviderRequest(mobile_apis::FunctionID::GetAppServiceDataID,
+                      hmi_apis::FunctionID::AppService_GetAppServiceData,
+                      &(*message_),
+                      true);
+}
+
+void GetAppServiceDataRequest::on_event(
+    const event_engine::MobileEvent& event) {
+  const smart_objects::SmartObject& event_message = event.smart_object();
+
+  auto msg_params = event_message[strings::msg_params];
+  SendResponse(true, mobile_apis::Result::SUCCESS, NULL, &msg_params);
```

Comment:
Response should be based on provider response. You can use my PR as an example: https://github.com/smartdevicelink/sdl_core/pull/2815/files#diff-0d2998ed0f4c24623a191ebe9ca8742fR168